src/mlpack/methods/reinforcement_learning/q_learning.hpp

```diff
@@ -78,7 +78,7 @@ class QLearning
    * @param environment Reinforcement learning task.
    */
   QLearning(TrainingConfig config,
-            NetworkType network,
+            NetworkType& network,
```

Comment:
I think it makes sense to use a reference for the other parameters as well, just to avoid any confusion, why we use move semantics for all the other parameters.

Reply:
Absolutely.